### ISSUE TITLE
Error: DNS2.php, line 900

### ISSUE THREAD

**Issue by gavilletloic** [mikepultz/netdns2#120]
https://github.com/mikepultz/netdns2/blob/master/Net/DNS2.php, line 900, master branch.

Phpstorm returns the error "Thrown object must be an instance of 'Exception' or 'Throwable'":900

It's this line:

` throw $this->last_exception;`

I didn't investigate, but I guess that the solution could be:

`throw new Net_DNS2_Exception($this->last_exception);` or `throw new Exception($this->last_exception);`

**Comment by jdwx** [comment]
Phpstorm wants the Net_DNS2::last_exception property to be type-hinted.

E.g., something like:

```
    /*
     * the last exception that was generated
     */
    public ?Net_DNS2_Exception $last_exception = null;
```

or:

```
    /**
     * the last exception that was generated
     * @var Net_DNS2_Exception|null
     */
    public $last_exception = null;
```

**Comment by mikepultz** [comment]
Sure- I added a typehint. Will be included in v1.5.3
